test/quick_test/test_images.py

```diff
@@ -0,0 +1,219 @@
+    # print("Image type: ", type(image))
+    # print("Image shape: ", image.shape)
+
+    return image
+
+
+def compute_dice_score(test_data, reference_data):
+    """
+    Compute the dice score.
+
+    Parameters
+    ----------
+    test_data : np.ndarray
+        Test image data.
+    reference_data : np.ndarray
+        Reference image data.
+
+    Returns
+    -------
+    np.ndarray
+        Dice score for each class.
+    """
+
+    # Labels
+    labels = [2, 3, 4, 5, 7, 8, 10, 11, 12, 13, 14, 15,
+              16, 17, 18, 24, 26, 28, 41, 42, 43, 44,
+              46, 47, 49, 50, 51, 52, 53, 54, 58, 60]
+
+    # Classes
+    classes = len(labels)
+    # classes = np.unique(test_data)
+    # print("Classes: ", classes)
+
+    dice_score = np.zeros(shape=classes)
+
+    for idx in range(classes):
+        current_label = labels[idx]
+        reference_sum = np.sum(reference_data == current_label)
+        test_sum = np.sum(test_data == current_label)
+        intersection = np.sum((reference_data == current_label) & (test_data == current_label))
+
+        if intersection == 0 and classes == 2:
+            if current_label == 17:
+                swap_lab = 53
+            else:
+                swap_lab = 17
+
+            reference_sum = np.sum(reference_data == swap_lab)
+            intersection = np.sum((reference_data == swap_lab) & (test_data == current_label))
+
+        dice_score[idx] = (2 * intersection) / (reference_sum + test_sum)
+
+    dice_score[np.isnan(dice_score)] = 0.
+
+    print("\nDice score: ", dice_score)
+
+    return dice_score
+
+
+def compute_mean_square_error(test_data, reference_data):
+    """
+    Calculate the mean square error.
+
+    Parameters
+    ----------
+    test_data : np.ndarray
+        Test image data.
+    reference_data : np.ndarray
+        Reference image data.
+
+    Returns
+    -------
+    float
+        Mean square error.
+    """
+
+    mse = ((test_data - reference_data) ** 2).mean()
+    print("\nMean square error: ", mse)
+
+    return mse
+
+
+def test_image_headers(sample_subjects_dir, sample_subject_id, sample_reference_id):
+    """
+    Test image headers using nibabel.
+
+    Parameters
+    ----------
+    sample_subjects_dir : str
+        Path to the sample subjects directory.
+    sample_subject_id : str
+        Sample subject id.
+    sample_reference_id : str
+        Sample reference id.
+    """
+
+    # Load images
+    test_image = load_image(sample_subjects_dir, sample_subject_id, "brain.mgz")
+    reference_image = load_image(sample_subjects_dir, sample_reference_id, "brain.mgz")
+
+    # Get the image headers
+    headers = [test_image.header, reference_image.header]
+
+    # Check the image header
+    assert nibabel.cmdline.diff.get_headers_diff(headers) == OrderedDict(), "Image headers do not match"
+    print("Image headers are correct")
+
+
+def test_seg_data(sample_subjects_dir, sample_subject_id, sample_reference_id):
```

Comment:
parameters should probably be subjects_dir, reference_subjects_dir and subect_id.
subject_id should then probably be a fixture with params.

We also talked about some general ideas (e.g. moving data to fixtures)

Reply:
the reference subjects are inside of subjects_dir